### PR TITLE
JSON cleanup of IoT Hub service direct method + unit tests

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/service/src/DirectMethod/Models/DirectMethodClientResponse.cs [hunk 1]
```diff
@@ -28,13 +28,13 @@ protected internal DirectMethodClientResponse()
...
-        /// Get the payload object. May be null or empty.
+        /// Get the payload as a JSON string.
         /// </summary>
         /// <remarks>
-        /// The payload can be null or primitive type (e.g., string, int/array/list/dictionary/custom type)
+        /// To get the payload as a specified type, use <see cref="TryGetPayload{T}(out T)"/>.
         /// </remarks>
         [JsonIgnore]
-        public object Payload => JsonConvert.DeserializeObject((string)JsonPayload.Value);
+        public string PayloadAsString => JsonPayload.Value<string>();
```
Comment (drwill-ms) [inline iothub/service/src/DirectMethod/Models/DirectMethodClientResponse.cs:37]: This doesn't work as well as we'd like, and it often exposes JArray and JObject types.

So, let's give them the serialized string and add a `TryGetValue<T>()` like we have done elsewhere.

File: iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs [hunk 2]
```diff
@@ -14,20 +14,25 @@ public class DirectMethodServiceRequest
...
+            MethodName= methodName;
         }
 
         /// <summary>
         /// The method name to run.
         /// </summary>
-        [JsonProperty("methodName", Required = Required.Always)]
-        public string MethodName { get; set; }
+        [JsonProperty("methodName")]
+        public string MethodName { get; }
```
Comment (drwill-ms) [inline iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs:30]: Now readonly.

File: iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs [hunk 3]
```diff
@@ -14,20 +14,25 @@ public class DirectMethodServiceRequest
...
 
         /// <summary>
         /// The payload to have serialized and send as JSON.
         /// </summary>
-        [JsonProperty("payload", Required = Required.AllowNull)]
```
Comment (drwill-ms) [inline iothub/service/src/DirectMethod/Models/DirectMethodServiceRequest.cs:30]: Dropping the required tags where not really necessary. We don't want to asser the service sends or doesn't send fields.

File: iothub/service/tests/DirectMethod/DirectMethodClientResponseTests.cs [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class DirectMethodClientResponseTests
+    {
+        [TestMethod]
+        public void DirectMethodClientResponse_Payload_DateTimeOffset()
+        {
+            // arrange
+
```
Comment (tmahmood-microsoft) [inline iothub/service/tests/DirectMethod/DirectMethodClientResponseTests.cs:21]: few extra lines in this file

**Review by tmahmood-microsoft — APPROVED** [review]

**Review by brycewang-microsoft — APPROVED** [review]

File: e2e/test/iothub/device/MethodE2ECustomPayloadTests.cs [hunk 5]
```diff
@@ -153,7 +152,8 @@ public async Task Method_DeviceReceivesMethodAndResponse_DictionaryPayload_Amqp(
 
             VerboseTestLogger.WriteLine($"{nameof(ServiceSendMethodAndVerifyResponseAsync)}: Method status: {methodResponse.Status}.");
             methodResponse.Status.Should().Be(200);
-            JsonConvert.SerializeObject(methodResponse.Payload).Should().BeEquivalentTo(JsonConvert.SerializeObject(response));
+            methodResponse.TryGetPayload(out T actual).Should().BeTrue();
+            JsonConvert.SerializeObject(actual).Should().BeEquivalentTo(JsonConvert.SerializeObject(response));
```
Comment (abhipsaMisra) [inline e2e/test/iothub/device/MethodE2ECustomPayloadTests.cs:156]: `JsonConvert.SerializeObject(actual).Should().Be(JsonConvert.SerializeObject(response));`?

String assertions can use `Should().Be(...)`

File: iothub/service/tests/DirectMethod/DirectMethodClientResponseTests.cs [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.Devices.Tests.DirectMethod
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class DirectMethodClientResponseTests
```
Comment (abhipsaMisra) [inline iothub/service/tests/DirectMethod/DirectMethodClientResponseTests.cs:15]: 👍🏽

**Review by abhipsaMisra — COMMENTED** [review]
lgtm!

File: e2e/test/iothub/device/MethodE2ECustomPayloadTests.cs (hunk 5 above)
Comment (drwill-ms) [reply e2e/test/iothub/device/MethodE2ECustomPayloadTests.cs:156]: In this case, it is of type T and I don't want to special case strings.

**Pull request merged**